spring-cloud-starter-stream-app-function/src/test/java/org/springframework/cloud/stream/app/function/app/FunctionAppConfigurationTests.java

```diff
@@ -77,7 +77,7 @@ public void test() throws Exception {
         public void test() throws Exception {
             processor.input().send(MessageBuilder.withPayload("hello").build());
             Message<?> received = messageCollector.forChannel(processor.output()).poll(1, TimeUnit.SECONDS);
-            assertThat(received.getPayload(), Matchers.is("hello".length()));
+            assertThat(Integer.valueOf(received.getPayload().toString()), Matchers.is("hello".length()));
```

Comment:
Can we just do 
```
assertThat(((byte[])received.getPayload()).length, Matchers.is("hello".length()));
```

Reply:
Hey, we cant. The received payload is `String`, and in that case, the `LengthCounter` function is under the test, so it returns the length of the given string. For some reasons (possible changes in 2.x version) message controller returns unconverted value, so we have to convert it manually to `Integer` in that case.

Reply:
I may need to run it but if the received payload is a `String` then why do we need `toString()`?
Also, in 2.x the returned payload should always be `byte[]`

Reply:
Strange, because return type is `String`